geomstats/learning/frechet_mean.py

```diff
@@ -333,6 +333,50 @@ def _adaptive_gradient_descent(
 
     return current_mean
 
+def _circle_mean (points):
+    """Determine the mean on a circle.
+    
+    Data are expected in radians in the range [-pi, pi). The mean is returned
+    in the same range. If the mean is unique, this algorithm is guaranteed to
+    find it. It is not vulnerable to local minima of the Frechet function. If
+    the mean is not unique, the algorithm only returns one of the means. Which
+    mean is returned depends on numerical rounding errors.
+    
+    Reference
+    ---------
+    Hotz, T. and S. F. Huckemann (2015), "Intrinsic means on the circle:
+    Uniqueness, locus and asymptotics", Annals of the Institute of Statistical
+    Mathematics 67 (1), 177–193.
+    https://arxiv.org/abs/1108.2141
+    """
+    sample_size = points.size
+    mean0 = gs.mean(points)
+    var0 = gs.var(points)
+    sorted_points = gs.sort(points)
+    means = _circle_variances(mean0, var0, sample_size, sorted_points)
+    return means[gs.argmin(means[:,1]),0]
+
+def _circle_variances (mean0, var0, sample_size, points):
+    n = sample_size
+    means = (mean0 + gs.linspace(0, 2 * gs.pi, n, endpoint=False)) % (2 * gs.pi)
+    means[means >= gs.pi] -= 2 * gs.pi
+    parts = [(sum(points) / n) if means[0] < 0 else 0]
+    m_plus = means >= 0
+    left_sums = gs.cumsum(points)
+    right_sums = left_sums[-1] - left_sums    
+    i = gs.array(range(n))
+    j = i[1:]
+    parts2 = right_sums[:-1] / (n-j)
+    parts2[m_plus[1:]] = (left_sums[:-1] / j)[m_plus[1:]]
+    parts = gs.hstack([parts, parts2])
+    #Formula (6) from Hotz, Huckemann
```

Comment:
Could you add the reference in the docstring too, as you did for `_circle_mean`? It can make this function more self-consistent, for users who would want details.